apps/src/blockly/addons/cdoConnectionChecker.js

```diff
@@ -0,0 +1,58 @@
+import GoogleBlockly from 'blockly/core';
+import {customConnectionBlockTypes} from './cdoConstants';
+
+/**
+ * A connection checker that imposes stricter typing rules for CDO custom
+ * connection type checks, but uses the same rules for dragging and safety.
+ * This checker still expects nullable arrays of string for connection
+ * type checks, and still looks for intersections in the arrays. Unlike the
+ * default checker, check arrays that contain a custom connection type are
+ * only compatible with other check arrays with the same connection type.
+ * @implements {Blockly.IConnectionChecker}
+ */
+
+export default class CdoConnectionChecker extends GoogleBlockly.ConnectionChecker {
+  /**
+   * Constructor for the connection checker.
+   */
+  constructor() {
```

Comment:
Do we need this constructor?

Reply:
I used the strict connection checker plugin as a guide, which also included this. However, it looks like it works just fine without it, so I've taken it out.

For reference: https://github.com/google/blockly-samples/blob/master/plugins/strict-connection-checker/src/index.js#L26-L28